metal/loadbalancers/emlb/emlb.go

```diff
@@ -10,24 +10,62 @@ import (
 )
 
 type LB struct {
-	client *lbaas.APIClient
+	client     *lbaas.APIClient
+	controller *controller
 }
 
 func NewLB(k8sclient kubernetes.Interface, config string) *LB {
+	metalAPIKey := "TODO"
 	lb := &LB{}
-	emlbConfig := lbaas.NewConfiguration()
-	lb.client = lbaas.NewAPIClient(emlbConfig)
+	lb.controller = NewController(metalAPIKey)
 	return lb
 }
 
 func (l *LB) AddService(ctx context.Context, svcNamespace, svcName, ip string, nodes []loadbalancers.Node) error {
```

Comment:
These arguments probably aren't sufficient/applicable for this case:
- `svcNamespace` useful here
- `svcName` also useful
- `ip` might be useful (we need to figure out what IP address it is)
- `nodes` seems very BGP-focused at the moment

What we need here is:
- The NodePort (for first pass we will use the same port on the LB, so if NodePort is 8000 we use 8000 on LB)
- The public IPs of the nodes on which the service is running

Reply:
The IP should be handed to this function, having been determined in the EIP reconciliation.

The nodes list will generally be every node in the cluster because NodePorts are cluster-wide. Even if a node is not running a particular pod/service, kube-proxy will route connections received for that service to a node where the service is running.

For the purposes of a metro scoped loadbalancer, it may make sense to filter this list to the nodes the loadbalancer can reach within the same metro. Arguable and probably not a first-pass concern.

Reply:
> The IP should be handed to this function, having been determined in the EIP reconciliation.

Sound's like this current behavior  may need to be circumvented for this service implementation.